frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,19 @@
+import { AppRoute, DataStatus } from '~/libs/enums/enums.js';
+import { useAppSelector, useEffect, useNavigate } from '~/libs/hooks/hooks.js';
+
+type Properties = {
+  children: React.ReactNode;
+};
+const ProtectedRoute: React.FC<Properties> = ({ children }: Properties) => {
+  const { auth } = useAppSelector((users) => users);
+  const navigate = useNavigate();
+  useEffect(() => {
+    if (auth.dataStatus !== DataStatus.FULFILLED) {
```

Comment:
you need to check `user` on existence, not `dataStatus` equality

Reply:
How can i do it? I need compare something with something other. What i should take?

Reply:
If it's value is `null` (there is no authorized user) then it's falsy value and on if check it would behave as falsy value